src/base/utils/version.h

```diff
@@ -58,7 +58,7 @@ namespace Utils
 
         template <typename ... Other>
         constexpr Version(Other ... components)
-            : m_components {{components ...}}
+            : m_components {{static_cast<T>(components) ...}}
```

Comment:
Could you explain what does this mean, please?

Reply:
Version {1,2,3} will be deduced as [int, int, int], it neglects the fact that `T` could be other smaller type (uint8_t) and will issue an narrow conversion warning, hence a type conversion here.

Reply:
Just for fun:
```diff
diff --git a/src/base/utils/version.h b/src/base/utils/version.h
index 19213cb35..6fcc26425 100644
--- a/src/base/utils/version.h
+++ b/src/base/utils/version.h
@@ -30,7 +30,9 @@
 #define QBITTORRENT_UTILS_VERSION_H
 
 #include <array>
+#include <initializer_list>
 #include <stdexcept>
+#include <utility>
 
 #include <QDebug>
 #include <QString>
@@ -56,9 +58,8 @@ namespace Utils
 
         constexpr Version(const ThisType &other) = default;
 
-        template <typename ... Other>
-        constexpr Version(Other ... components)
-            : m_components {{components ...}}
+        constexpr Version(std::initializer_list<T> components)
+            : m_components {makeArray(components, std::make_index_sequence<N>())}
         {
         }
 
@@ -159,6 +160,12 @@ namespace Utils
     private:
         using ComponentsArray = std::array<T, N>;
 
+        template <std::size_t ...Inds>
+        constexpr static ComponentsArray makeArray(std::initializer_list<T> l, std::index_sequence<Inds...>)
+        {
+            return ComponentsArray{*(l.begin() + Inds) ... };
+        }
+
         template <typename StringsList>
         static ComponentsArray parseList(const StringsList &versionParts)
         {
```
With C++14 the `std::make_index_sequence()` call would use `components.size()`, but in C++11 the `size()` function is not `constexpr`. I post hoping somebody can find a solution without adding  `throw`.

Reply:
Maybe an (optional?) C++14 support in the source base makes sense? As far as I understand, the show stopper is Ubuntu 14.04 (end of life next April) with its GCC 4.x. But how many Ubuntu 14.04 users want to build packages by themselves and do not have an updated toolchain? IMO, a negligible part, so dropping GCC 4 support would not make any significant harm.